qiita_pet/handlers/rest/study_samples.py

```diff
@@ -0,0 +1,81 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+from tornado.escape import json_encode
+
+from qiita_db.handlers.oauth2 import authenticate_oauth
+from .rest_handler import RESTHandler
+
+
+class StudySamplesHandler(RESTHandler):
+
+    @authenticate_oauth
+    def get(self, study_id):
+        study = self.study_boilerplate(study_id)
+        if study is None:
+            return
+
+        if study.sample_template is None:
+            samples = []
+        else:
+            samples = list(study.sample_template.keys())
+
+        self.write(json_encode(samples))
+        self.finish()
+
+
+class StudySamplesCategoriesHandler(RESTHandler):
+
+    @authenticate_oauth
+    def get(self, study_id, categories):
+        if not categories:
+            self.fail('No categories specified', 405)
+            return
+
+        study = self.study_boilerplate(study_id)
+        if study is None:
+            return
+
+        categories = categories.split(',')
+
+        if study.sample_template is None:
+            self.fail('Category not found', 404)
```

Comment:
The ID is already implied in the URI itself, so it seems a bit redundant. Also, after modeling from GitHub's API, none of their messages seemed to imply any specific information about the requested resource, they were succinct and generic.